KafkaProducerConfig is missing if requested from a KafkaListener

The following issue exists in Version 2.1.0 of micronaut kafka. It does not exist in version 2.1.0.RC1. 
We think that this problem was introduced with https://github.com/micronaut-projects/micronaut-kafka/commit/0561addba7a0f7366b0aa6a0fd735f8f180bc064

In our case it causes the condition here to be false and the else branch to be executed:

https://github.com/micronaut-projects/micronaut-core/blob/7d2c721cd429dedeedfde3a13076b72714ae4c17/inject/src/main/java/io/micronaut/context/DefaultBeanContext.java#L2354-L2361

(The KafkaClientScope is missing)

### Task List

- [X] Steps to reproduce provided
- [X] Stacktrace (if present) provided
- [X] Example that reproduces the problem uploaded to Github
- [X] Full description of the issue provided (see below)

### Steps to Reproduce

1. Create a Micronaut Project
2. Create a KafkaListener
3. Create a KafkaClient (KafkaProducer)
4. Reference the KafkaClient inside the KafkaListener so that it should be injected

### Expected Behaviour

Application starts without errors, beans are instatiated and configured. (Worked with micronaut-kafka 2.1.0.RC1, broken with micronaut-kafka 2.1.0)

### Actual Behaviour

The KafkaProducerConfiguration bean is missing and we get a BeanInstatiationException

```
11:05:58.439 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [de.ksmwsk.KafkaConsumer] could not be loaded: Error instantiating bean of type  [de.ksmwsk.KafkaProducer]

Message: Missing bean argument [AbstractKafkaProducerConfiguration producerConfiguration] for type: org.apache.kafka.clients.producer.KafkaProducer. Required arguments: AbstractKafkaProducerConfiguration producerConfiguration
Path Taken: new KafkaConsumer([KafkaProducer adapter]) --> new KafkaProducer([Producer kafkaProducer])
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [de.ksmwsk.KafkaConsumer] could not be loaded: Error instantiating bean of type  [de.ksmwsk.KafkaProducer]

Message: Missing bean argument [AbstractKafkaProducerConfiguration producerConfiguration] for type: org.apache.kafka.clients.producer.KafkaProducer. Required arguments: AbstractKafkaProducerConfiguration producerConfiguration
Path Taken: new KafkaConsumer([KafkaProducer adapter]) --> new KafkaProducer([Producer kafkaProducer])
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1533)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:220)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:2763)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:227)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:166)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:64)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:294)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:280)
	at de.ksmwsk.KafkaApplication.main(KafkaApplication.java:8)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [de.ksmwsk.KafkaProducer]

Message: Missing bean argument [AbstractKafkaProducerConfiguration producerConfiguration] for type: org.apache.kafka.clients.producer.KafkaProducer. Required arguments: AbstractKafkaProducerConfiguration producerConfiguration
Path Taken: new KafkaConsumer([KafkaProducer adapter]) --> new KafkaProducer([Producer kafkaProducer])
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1859)
	at io.micronaut.context.DefaultBeanContext.getScopedBeanForDefinition(DefaultBeanContext.java:2386)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2301)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2273)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1233)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1035)
	at de.ksmwsk.$KafkaProducerDefinition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1888)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2630)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2616)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2299)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2273)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1233)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1035)
	at de.ksmwsk.$KafkaConsumerDefinition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1888)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2630)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2177)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1531)
	... 8 common frames omitted
```

### Environment Information

- **Operating System**: Windows 10
- **Micronaut Version:** 2.0.0
- **JDK Version:** 1.8

### Example Application

- https://github.com/ckosmowski/micronaut-examples/tree/issues/kafka-producer-config
(Just try to start KafkaApplication).

--- a/kafka-streams/src/test/resources/logback.xml
+++ b/kafka-streams/src/test/resources/logback.xml
@@ -12,6 +12,6 @@
         <appender-ref ref="STDOUT" />
     </root>
 <!--    <logger name="io.micronaut.context" level="TRACE"/>-->
-<!--    <logger name="io.micronaut.configuration.kafka.embedded" level="TRACE" />-->
-<!--    <logger name="com.testcontainers" level="INFO" />-->
+    <logger name="io.micronaut.configuration.kafka.embedded" level="TRACE" />
+    <logger name="com.testcontainers" level="INFO" />
 </configuration>

--- a/kafka/src/main/java/io/micronaut/configuration/kafka/annotation/KafkaListener.java
+++ b/kafka/src/main/java/io/micronaut/configuration/kafka/annotation/KafkaListener.java
@@ -35,7 +35,6 @@ import static java.lang.annotation.RetentionPolicy.RUNTIME;
 @Retention(RUNTIME)
 @Target({ElementType.TYPE, ElementType.METHOD, ElementType.ANNOTATION_TYPE})
 @MessageListener
-@DefaultScope(Context.class)
 public @interface KafkaListener {
 
     /**

--- a/kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java
+++ b/kafka/src/main/java/io/micronaut/configuration/kafka/processor/KafkaConsumerProcessor.java
@@ -26,7 +26,6 @@ import io.micronaut.configuration.kafka.exceptions.KafkaListenerException;
 import io.micronaut.configuration.kafka.exceptions.KafkaListenerExceptionHandler;
 import io.micronaut.configuration.kafka.serde.SerdeRegistry;
 import io.micronaut.context.BeanContext;
-import io.micronaut.context.annotation.Context;
 import io.micronaut.context.annotation.Requires;
 import io.micronaut.context.processor.ExecutableMethodProcessor;
 import io.micronaut.core.annotation.AnnotationValue;
@@ -87,7 +86,6 @@ import java.util.regex.Pattern;
  */
 @Singleton
 @Requires(beans = KafkaDefaultConfiguration.class)
-@Context
 public class KafkaConsumerProcessor
         implements ExecutableMethodProcessor<KafkaListener>, AutoCloseable, ConsumerRegistry {
 
@@ -141,6 +139,20 @@ public class KafkaConsumerProcessor
         this.executorScheduler = Schedulers.from(executorService);
         this.producerRegistry = producerRegistry;
         this.exceptionHandler = exceptionHandler;
+        this.beanContext.getBeanDefinitions(Qualifiers.byType(KafkaListener.class))
+                        .forEach(definition -> {
+                            // pre-initialize singletons before processing
+                            if (definition.isSingleton()) {
+                                try {
+                                    beanContext.getBean(definition.getBeanType());
+                                } catch (Exception e) {
+                                    throw new MessagingSystemException(
+                                            "Error creating bean for @KafkaListener of type [" + definition.getBeanType() + "]: " + e.getMessage(),
+                                            e
+                                    );
+                                }
+                            }
+                        });
     }
 
     @Nonnull

